SlidePuzzle/src/main/java/ThirdApproach/Movement.java

```diff
@@ -0,0 +1,46 @@
+package ThirdApproach;
+
+import static ThirdApproach.Puzzle.*;
+
+public class Movement {
+    static int slv = 0;
+    public static boolean makeMove(char key) {
+        int tmp;
+
+        switch (key) {
```

Comment:
I think you can refactor this part :) because right now it is broking the "[Open Closed Principle](https://javatechonline.com/solid-principles-the-open-closed-principle/)", In classes we talked a little bit about "Command" pattern, so, an idea to improve this part could be:
`interface ControlCommand {
    void execute();
}
class WestControlCommand implements ControlCommand {
   public void execute() {
      // do something
   }
}

class CommandHandler {
  private Map<String, ControlCommand> controlCommandMap;
  
  public void handlerCommandRequest(String action) {
      ControlCommand control = controlCommandMap.get(action);
      control.execute();
  }
}
`